dynatrace_topology/stackstate_checks/dynatrace_topology/dynatrace_topology.py

```diff
@@ -0,0 +1,304 @@
+                tag_label += '[{0}]'.format(tag['context'])
+            if tag.get('key'):
+                tag_label += "{0}".format(tag['key'])
+            if tag.get('value'):
+                tag_label += ":{0}".format(tag['value'])
+            tags.append(tag_label)
+        return tags
+
+    def get_labels(self, item):
+        """
+        Extract labels and tags for each component
+        :param item: the component item
+        :return: the list of added labels for a component
+        """
+        labels = []
+        # append management zones in labels for each existing component
+        for zone in item.get("managementZones", []):
+            if zone.get("name"):
+                labels.append("managementZones:{}".format(zone.get("name")))
+        if item.get("entityId"):
+            labels.append(item["entityId"])
+        if "monitoringState" in item:
+            actual_state = item["monitoringState"].get("actualMonitoringState")
+            expected_state = item["monitoringState"].get("expectedMonitoringState")
+            if actual_state:
+                labels.append("actualMonitoringState:{}".format(actual_state))
+            if expected_state:
+                labels.append("expectedMonitoringState:{}".format(expected_state))
+        for technologies in item.get("softwareTechnologies", []):
+            tech_label = ''
+            if technologies.get('type'):
+                tech_label += technologies['type']
+            if technologies.get('edition'):
+                tech_label += ":{}".format(technologies['edition'])
+            if technologies.get('version'):
+                tech_label += ":{}".format(technologies['version'])
+            labels.append(tech_label)
+        labels_from_tags = self.get_labels_from_dynatrace_tags(item)
+        labels.extend(labels_from_tags)
+        # prefix the labels with `dynatrace-` for all labels
+        labels = ["dynatrace-{}".format(label) for label in labels]
+        return labels
+
+    def get_dynatrace_json_response(self, endpoint, timeout=10):
+        """
+        Make a request to Dynatrace endpoint through session
+        :param endpoint: Dynatrace API Endpoint to call
+        :param timeout: timeout for a response
+        :return: Response of each endpoint
+        """
+        headers = {"Authorization": "Api-Token {}".format(self.token)}
+        resp = None
+        msg = None
+        self.log.debug("URL is {}".format(endpoint))
+        try:
+            with Session() as session:
+                session.headers.update(headers)
+                session.verify = self.verify
+                if self.cert:
+                    session.cert = (self.cert, self.keyfile)
+                resp = session.get(endpoint)
+            if resp.status_code != 200:
+                raise Exception("Got %s when hitting %s" % (resp.status_code, endpoint))
+            return yaml.safe_load(resp.text)
```

Comment:
Please move inside the `with`. This will ensure that if `resp.text` hits an exception that the session will be closed.